File: Demos/Datasnap/Server/proxy/java_android/com/embarcadero/javaandroid/TDBXStringValue.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javaandroid;

/**
 * 
 * Wraps the String type and allows it to be null
 *
 */

public class TDBXStringValue extends DBXValue{
	protected boolean ValueNull = false;
	private String DBXStringValue;	
	
	public TDBXStringValue() {
		super();
		setDBXType(DBXDataTypes.WideStringType);
	}
	
	public void setNull() {
		ValueNull = true;
		DBXStringValue = "";
	}

	public boolean isNull() {
		return ValueNull;
	}
	
	@Override
	public void SetAsString(String Value) throws DBXException {
		ValueNull = false;
		DBXStringValue = Value;
	}
	
	@Override
	public String GetAsString() throws DBXException {
		return DBXStringValue;
	}
}

File: Demos/Datasnap/Server/proxy/objectivec_ios81/Proxy/TDBXStringValue.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TDBXStringValue.h"


@implementation TDBXStringValue
-(id)init{
	self = [super init];
	if (self) {
		[self setDBXType:WideStringType];
	}
	return self;
}
-(void) SetNull {
	ValueNull = YES;
	DBXStringValue = @"";
}

-(bool) isNull {
	return ValueNull;
}

-(void) SetAsString:(NSString*) value{
	ValueNull = NO;
	DBXStringValue = value;
}

-(NSString*) GetAsString {
	return DBXStringValue;
}

@end
